tests/packages/sharedlib-in-package/mypkg/__init__.py

```diff
@@ -0,0 +1,55 @@
+# SPDX-FileCopyrightText: 2024 The meson-python developers
+#
+# SPDX-License-Identifier: MIT
+
+import os
+import sys
+
+
+# Note: the below function is used in the docs via a literalinclude directive.
+#       When making any changes to this file, please ensure the right set of
+#       lines is pulled into the docs, and adjust the :lines: attribute of
+#       the literalinclude if needed.
+
+
+def _enable_sharedlib_loading():
+    """
+    Ensure the shared libraries in this dir and the `sub` subdir can be loaded
+    on Windows.
+
+    One shared library is installed alongside this ``__init__.py`` file.
+    Windows can load it because it searches for DLLs in the directory a ``.pyd``
+    (Python extension module) is located in. Cygwin does not though. For a
+    shared library in another directory inside the package, Windows also needs
+    a hint.
+
+    This function is Windows-specific due to lack of RPATH support on Windows.
+    It cannot find shared libraries installed within wheels unless we either
+    amend the DLL search path or pre-load the DLL.
```

Comment:
I find this docstring a bit repetitive and contrived in places. I think it can be made much more concise. I can try to come up with a better formulation, if you like.

Reply:
Yes please, that could make it better.